collect/collect.go

```diff
@@ -246,9 +232,9 @@ func (i *InMemCollector) collect() {
 
 		// ok, the peer queue is low enough, let's wait for new events from anywhere
 		select {
-		case sig, ok := <-i.toSend:
-			if ok {
-				i.send(sig.trace)
+		case <-i.sendTicker.C:
```

Comment:
Why are we only checking the sendTicker in this select but not the others?

Reply:
A very good question, I'm just pondering this now 🤔

Reply:
I think this should go up in the previous `select` loop too. The multiple `select`s here with different options are to prioritize getting traffic _out of_ the system above accepting new traffic _into_ the system, so expunging traces that are ready to be sent feels like a good thing to do in the first loop through.

Reply:
Looking at `sendASAP` and realizing it can be triggered during peer span processing makes me believe more strongly that checking the send ticker should be in both places. If we're behind and reading a bunch of peer spans, the probability we'll overflow the trace buffer also increases. If `sendASAP` is triggering a goroutine for every ejected trace, falling behind could trigger a flurry of extra goroutines, increasing CPU, making us fall further behind. Womp womp. Moving this ticker check above will help keep the trace ring clean, reducing the chance of ejected traces, and minimizing calls to `sendASAP`.

If you decide on the non-goroutine based method for handling `sendASAP`, then I'm once again not sure if it's useful to put this in the upper `select`. Wheee!